rust/relay/src/main.rs

```diff
@@ -106,13 +107,14 @@ async fn main() -> Result<()> {
         args.highest_port,
     );
 
-    let channel = if let Some(token) = args.portal_token.as_ref() {
+    let channel = if let Some(token) = args.portal_token.clone() {
```

Comment:
```suggestion
    let channel = if let Some(token) = args.portal_token.map(SecretString::new).as_ref() {
```

Ideally, we would somehow teach `clap` that it should directly convert the token into a `SecretString`.

Reply:
Hmm that's actually quite easy to do, but passing it to `connect_to_portal` wasn't so straightforward because `args` is passed too and already has a reference. I'm not super proud of how I worked around the issue (checking portal_token `is_some()` twice). Open to suggestions.

Reply:
portal_token can't be cloned or copied once it's an `Option<SecretString>` which makes it a little trickier.

Reply:
I pushed two commits for you that fix this :)